packages/client-purpose-updater/src/clientPurposeUpdaterConsumerServiceV2.ts

```diff
@@ -0,0 +1,59 @@
+import {
+  CorrelationId,
+  generateId,
+  PurposeEventEnvelopeV2,
+} from "pagopa-interop-models";
+import { match } from "ts-pattern";
+import {
+  getInteropHeaders,
+  RefreshableInteropToken,
+} from "pagopa-interop-commons";
+import { getInteropBeClients } from "./clients/clientsProvider.js";
+
+const { authorizationClient } = getInteropBeClients();
+
+export async function handleMessageV2(
+  message: PurposeEventEnvelopeV2,
+  refreshableToken: RefreshableInteropToken
+): Promise<void> {
+  await match(message)
+    .with({ type: "PurposeArchived" }, async (purposeMsg) => {
+      const token = (await refreshableToken.get()).serialized;
+      const maybePurposeId = purposeMsg.data.purpose?.id;
+
+      if (!maybePurposeId) {
+        return;
+      }
+
+      const purposeId = maybePurposeId;
+
+      await authorizationClient.client.removePurposeFromClients(undefined, {
+        params: { purposeId },
```

Comment:
```suggestion
      if (!purposeMsg.data.purpose) {
        throw missingKafkaMessageDataError("purpose", purposeMsg.type);
      }

      await authorizationClient.client.removePurposeFromClients(undefined, {
        params: { purposeId: purposeMsg.data.purpose.id },
```
I think it's better to throw in this case since we miss the purpose data

Reply:
I'm not sure if it should throw in case of error, that's why I asked if it would be better to put it in a `try catch`

At @Carminepo2 's suggestion the tests were not implemented, as it is only a call to a route that generates a new event, the consumer who intercepts that event will then do the rest

Reply:
> I'm not sure if it should throw in case of error, that's why I asked if it would be better to put it in a `try catch`
> 
> At @Carminepo2 's suggestion the tests were not implemented, as it is only a call to a route that generates a new event, the consumer who intercepts that event will then do the rest

The try catch you proposed was on the calling of the route, which is different from this case; I don't think the catch on the route is necessary but the purpose data should always be defined IMHO